src/Internal/Payment/Transition.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Class Transition
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment;
+
+/**
+ * A Transition object used for logging.
+ * Contains the details of a state transition during the payment process.
+ */
+class Transition {
+
+	/**
+	 * Order ID.
+	 *
+	 * @var string
+	 */
+	private $order_id;
+
+	/**
+	 * State of the transition.
+	 *
+	 * @var string
+	 */
+	private $from_state;
+
+	/**
+	 * State of the transition.
+	 *
+	 * @var string
+	 */
+	private $to_state;
+
+	/**
+	 * Changes made during the transition.
+	 *
+	 * @var array
+	 */
+	private $changes;
+
+	/**
+	 * Timestamp of the transition.
+	 *
+	 * @var int
+	 */
+	private $timestamp;
+
+	/**
+	 * Constructs the class, with changes, order ID, timestamp and state.
+	 *
+	 * @param string $order_id Order ID.
+	 * @param string $from_state State of the transition.
+	 * @param string $to_state State of the transition.
+	 * @param array  $changes Changes made during the transition.
+	 * @param int    $timestamp Timestamp of the transition.
+	 */
+	public function __construct( string $order_id, ?string $from_state, ?string $to_state, array $changes, int $timestamp ) {
```

Comment:
Would we ever provide `to_state` and `changes` within the constructor?

Reply:
Simplified the constructor